runners/direct-java/src/main/java/org/apache/beam/runners/direct/ParDoEvaluator.java

```diff
@@ -45,7 +46,7 @@
       DirectStepContext stepContext,
       CommittedBundle<InputT> inputBundle,
       AppliedPTransform<PCollection<InputT>, ?, ?> application,
-      Object fn,
+      Serializable fn, // may be OldDoFn or DoFn
```

Comment:
I believe the only `OldDoFn` is now GABW, so this should go away once `ReduceFn` is executed directly (a runners-core concept only)